ast/compile.go

```diff
@@ -942,6 +976,18 @@ func parseSchema(schema interface{}) (types.Type, error) {
 			return types.NewArray(nil, types.A), nil
 		}
 	}
+
+	// Assume types if not specified in schema
+	if len(subSchema.PropertiesChildren) > 0 {
+		if err := subSchema.Types.Add("object"); err == nil {
```

Comment:
So `err != nil` is nothing to worry about here and below? 🤔 I don't fully understand what's going on here, so this is on me 😅

Reply:
To clarify, this is to handle the case if the user uses one of the schema keywords (like `properties` or `items`) but didn't specify the `type` of the subschema (compare line 13 and 20 [here](https://github.com/jchen10500/anyOf-example/blob/ex-1/schemas/input_extend.json) for example, which can occur because JSON Schema allows "[factoring schemas](https://json-schema.org/understanding-json-schema/reference/combining.html#factoring-schemas)"). 

I think by this point since the `if` statement above have already checked that `type` is empty, then the internal `gojsonschema` library shouldn't return an error. I can certainly handle the error just in case :)